core/src/subject/SDNADecorators.ts

```diff
@@ -74,15 +74,43 @@ interface PropertyOptions {
     through: string;
     initial?: string,
     required?: boolean,
-    resolve?: boolean,
+    writable?: boolean,
+    resolveLanguage?: string;
 }
 export function subjectProperty(opts: PropertyOptions) {
     return function <T>(target: T, key: keyof T) {
         target["__properties"] = target["__properties"] || {};
         target["__properties"][key] = target["__properties"][key] || {};
         target["__properties"][key] = { ...target["__properties"][key], ...opts }
+
+        if (opts.writable) {
+            const value = key as string
+            target[`set${capitalize(value)}`] = () => {}
+        }
+
         Object.defineProperty(target, key, {});
-        //return descriptor;
+    };
+}
+
+interface FlagOptions {
+    through: string;
+    value: string;
+}
+export function subjectFlag(opts: FlagOptions) {
+    return function <T>(target: T, key: keyof T) {
+        target["__properties"] = target["__properties"] || {};
+        target["__properties"][key] = target["__properties"][key] || {};
+        target["__properties"][key] = { 
+            ...target["__properties"][key],
+            through: opts.through,
+            required: true,
+            initial: opts.value
```

Comment:
Hm, don't we need to store that it is a flag, which should have the consequence that the instance check not only checks for any link with the `through` predicate, but also that the target is `value`?